crates/loam-cli/src/commands/build/clients.rs

```diff
@@ -254,94 +259,115 @@ export default new Client.Client({{
         &self,
         workspace_root: &std::path::Path,
         contracts: Option<&Map<Box<str>, env_toml::Contract>>,
+        package_names: Vec<String>,
         network: &Network,
     ) -> Result<(), Error> {
-        let Some(contracts) = contracts else {
+        if package_names.is_empty() {
             return Ok(());
-        };
-        for (name, settings) in contracts {
-            if settings.workspace {
+        }
+        // ensure contract names are valid
+        if let Some(contracts) = contracts {
+            for (name, _) in contracts.iter().filter(|(_, settings)| settings.workspace) {
                 let wasm_path = workspace_root.join(format!("target/loam/{name}.wasm"));
                 if !wasm_path.exists() {
                     return Err(Error::BadContractName(name.to_string()));
                 }
-                eprintln!("📲 installing {name:?} wasm bytecode on-chain...");
-                let hash = cli::contract::install::Cmd::parse_arg_vec(&[
-                    "--wasm",
-                    wasm_path
-                        .to_str()
-                        .expect("we do not support non-utf8 paths"),
-                ])?
-                .run_against_rpc_server(None, None)
-                .await?
-                .into_result()
-                .expect("no hash returned by 'contract install'")
-                .to_string();
-                eprintln!("    ↳ hash: {hash}");
-
-                // Check if we have an alias saved for this contract
-                let alias = Self::get_contract_alias(name)?;
-                if let Some(contract_id) = alias {
-                    match self
-                        .contract_hash_matches(&contract_id, &hash, network)
-                        .await
-                    {
-                        Ok(true) => {
-                            eprintln!("✅ Contract {name:?} is up to date");
-                            continue;
-                        }
-                        Ok(false) if self.loam_env(LoamEnv::Production) == "production" => {
-                            return Err(Error::ContractUpdateNotAllowed(name.to_string()));
-                        }
-                        Ok(false) => eprintln!("🔄 Updating contract {name:?}"),
-                        Err(e) => return Err(e),
-                    }
+            }
+        }
+        for name in package_names {
+            let settings = match contracts {
+                Some(contracts) => contracts.get(&name as &str),
+                None => None,
+            };
+            // Skip only if contract is found and its workspace setting is false
+            if let Some(c) = settings {
+                if !c.workspace {
+                    eprintln!("Skipping {name:?} as its workspace setting is false");
```

Comment:
Haven't looked at the full output here, but my guess is that it will be annoying to see output for a no-op every single time. Do you think it would make sense to just not output anything, if this setting is `false`?

Reply:
Sounds good, will remove